### DISCUSSION TITLE
Adding more info about `framework` key in main.js of .storybook in the Storybook docs

### DISCUSSION THREAD

**Discussion by qur786** [storybookjs/storybook#18102] [Documentation] [answered]
In my company we are using storybook to build and test react components. So we have set it up in few of our repository, configuring the storybook through `sb` command line, it includes `framework` key in `main.js` with value as `@storybook/react`.
My concern is what does this `key and value` does behind the scene because in the absence of this field in `main.js` our storybook setup is working fine.
I have tried to find its related information in the `storybook docs` and tried to google about it, but couldn't find a valid explanation.
Can anybody from `storybook` team include more information about `framework` key field in the `main.js` file?

**Comment by shilman** [comment, marked as answer]
In Storybook 6.x and below, each framework (e.g. `@storybook/react`) shipped its own `start-storybook` and `build-storybook` binaries. These binaries filled in a generic "core" with framework-specific webpack & babel settings.

In Storybook 7, we are transitioning to a model where you run the storybook CLI directly using (`sb dev` / `sb build`). The framework-specific settings will be provided by the `framework` field in `.storybook/main.js`.

Starting in Storybook 6.4, `start-storybook`/`build-storybook` will use the new codepath if `framework` is set, and will use the old codepath if it's not. Functionally, the two modes should be identical. That said, we're currently designing Storybook 7, and we will likely change how this all works before it is released.

I hope that's a helpful explanation!
